internal/controlplane/handlers_providers.go

```diff
@@ -34,6 +38,43 @@ import (
 	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
 )
 
+func (s *Server) CreateProvider(ctx context.Context, req *minderv1.CreateProviderRequest) (*minderv1.CreateProviderResponse, error) {
+	entityCtx := engine.EntityFromContext(ctx)
+	projectID := entityCtx.Project.ID
+	provider := req.GetProvider()
+
+	var config json.RawMessage
+	if provider.Config != nil {
+		var marshallErr error
+
+		zerolog.Ctx(ctx).Debug().Str("config", provider.Config.String()).Msg("provider config")
+
+		config, marshallErr = provider.Config.MarshalJSON()
+		if marshallErr != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "error marshalling provider config: %v", marshallErr)
+		}
+	} else {
+		zerolog.Ctx(ctx).Debug().Msg("no provider config, will use default")
+	}
+
+	dbProv, err := s.providerManager.CreateFromConfig(ctx, db.ProviderClass(provider.GetClass()), projectID, provider.Name, config)
```

Comment:
This create rpc and the providerManager method are quite bare-bones and let the caller set up the auth token separately.

Reply:
The authentication flows should be part of the provider class, right?